### PR TITLE
process sorted RowChangedEvent in batch to increase throughput

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

optimization for https://github.com/pingcap/ticdc/issues/571

### What is changed and how it works?

- Emit RowChangedEvents in batch
- cache RowChangedEvent and flush them when reaching batch limit or receiving a ResolvedEvent

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- No release note

### CODE REVIEW

File: cdc/processor.go [hunk 1]
```diff
@@ -654,35 +660,66 @@ func (p *processor) sinkDriver(ctx context.Context) error {
...
-	defer log.Info("syncResolved stopped")
-	defer p.sinkEmittedResolvedReceiver.Stop()
-	metricWaitPrepare := waitEventPrepareDuration.WithLabelValues(p.changefeedID, p.captureID)
+	defer func() {
+		p.sinkEmittedResolvedReceiver.Stop()
+		log.Info("syncResolved stopped")
+	}()
+
+	events := make([]*model.PolymorphicEvent, 0, defaultSyncResolvedBatch)
+
```
Comment (leoppro) [inline cdc/processor.go:669]:
```suggestion
emitEvs := make([]*model.RowChangedEvent, 0, defaultSyncResolvedBatch)
```
and remove L670
and add `emitEvs = emitEvs[:0]` to L686

Comment (amyangfei) [reply cdc/processor.go:670]: addressed

**Pull request merged**